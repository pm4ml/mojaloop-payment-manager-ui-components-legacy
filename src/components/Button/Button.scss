@import '../../assets/styles/vars/colors.scss';
@import '../../assets/styles/vars/shadow.scss';
@import '../../assets/styles/vars/transitions.scss';

.input-button__input {
  outline: none;
  cursor: pointer;
  box-sizing: border-box;
  border: 0;
  display: inline-block;
  flex: 0 0 auto;
  white-space: nowrap;
  user-select: none;
  text-decoration: none;
  transition: $button__main-transition;
  font-style: normal;
  font-weight: 400;
  -webkit-font-smoothing: antialiased;
  border: 1px solid;
  align-items: center;
  color: $color__white;
  fill: $color__white;
  stroke: $color__white;
  border-bottom-width: 2px;

  box-shadow: $shadow-size--closed transparent;

  &:not(.noFill) {
    border-bottom-color: rgba(0, 0, 0, 0.2);
  }
  &:hover,
  &:focus {
    border-bottom-width: 2px;
    border-bottom-color: rgba(0, 0, 0, 0.2);
    &:active {
      border-bottom-color: transparent;
    }
  }

  &:active {
    color: $color__alpha-3;
    fill: $color__alpha-3;
    stroke: $color__alpha-3;
  }
}

.input-button__content {
  display: flex;
  align-items: center;
  justify-content: center;
}
.input-button__icon {
  display: flex;
  fill: inherit;
  stroke: inherit;
}
.input-button__icon--left {
  margin-right: 5px;
  margin-left: 0px;
}
.input-button__icon--right {
  margin-right: 0px;
  margin-left: 5px;
}

.input-button__mb-input--extra-small {
  padding: 2px 4px;
  font-size: 10px;
  height: 21px;
}
.input-button__mb-input--small {
  padding: 6px 10px;
  font-size: 11px;
  height: 27px;
}
.input-button__mb-input--medium {
  padding: 8px 13px;
  font-size: 12px;
  height: 33px;
}
.input-button__mb-input--large {
  padding: 10px 16px;
  font-size: 14px;
  height: 39px;
}

.input-button__mb-input--primary {
  background-color: $color__main-primary;
  border-color: $color__main-primary;

  &.noFill {
    background-color: transparent;
    color: $color__main-primary;
    fill: $color__main-primary;
    stroke: $color__main-primary;
  }

  &:active,
  &:focus {
    box-shadow: $shadow-size $shadow-color--primary;
  }

  &:hover,
  &:focus {
    background-color: $color__main-primary--active;
    border-color: $color__main-primary--active;

    &.noFill {
      color: $color__white;
      fill: $color__white;
      stroke: $color__white;
    }
  }
}
.input-button__mb-input--secondary {
  background-color: $color__main-secondary;
  border-color: $color__main-secondary;

  &.noFill {
    background: transparent;
    color: $color__main-secondary;
    fill: $color__main-secondary;
    stroke: $color__main-secondary;
  }

  &:active,
  &:focus {
    box-shadow: $shadow-size $shadow-color--secondary;
  }

  &:hover,
  &:focus {
    background-color: $color__main-secondary--active;
    border-color: $color__main-secondary--active;

    &.noFill {
      color: $color__white;
      fill: $color__white;
      stroke: $color__white;
    }
  }
}

.input-button__mb-input--success {
  background-color: $color__main-success;
  border-color: $color__main-success;
  color: $color__white;
  fill: $color__white;
  stroke: $color__white;

  &.noFill {
    background: transparent;
    color: $color__main-success;
    fill: $color__main-success;
    stroke: $color__main-success;
  }

  &:active,
  &:focus {
    box-shadow: $shadow-size $shadow-color--success;
  }

  &:hover,
  &:focus {
    background-color: $color__main-success--active;
    border-color: $color__main-success--active;

    &.noFill {
      color: $color__white;
      fill: $color__white;
      stroke: $color__white;
    }
  }
}
.input-button__mb-input--danger {
  background-color: $color__main-danger;
  border-color: $color__main-danger;
  color: $color__white;
  fill: $color__white;
  stroke: $color__white;

  &.noFill {
    background: transparent;
    color: $color__main-danger;
    fill: $color__main-danger;
    stroke: $color__main-danger;
  }

  &:active,
  &:focus {
    box-shadow: $shadow-size $shadow-color--danger;
  }

  &:hover,
  &:focus {
    background-color: $color__main-danger--active;
    border-color: $color__main-danger--active;

    &.noFill {
      color: $color__white;
      fill: $color__white;
      stroke: $color__white;
    }
  }
}

.input-button__mb-input--warning {
  background-color: $color__main-warning;
  border-color: $color__main-warning;

  &.noFill {
    background-color: transparent;
    color: $color__main-warning;
    fill: $color__main-warning;
    stroke: $color__main-warning;
  }

  &:active,
  &:focus {
    box-shadow: $shadow-size $shadow-color--warning;
  }

  &:hover,
  &:focus {
    background-color: $color__main-warning--active;
    border-color: $color__main-warning--active;

    &.noFill {
      color: $color__white;
      fill: $color__white;
      stroke: $color__white;
    }
  }
}

.input-button__mb-input--dark {
  background-color: $color__alpha-5;
  border-color: transparent;
  color: $color__white;
  fill: $color__white;
  stroke: $color__white;

  &.noFill {
    background-color: transparent;
    border-color: $color__alpha-5;
    color: $color__alpha-4;
    fill: $color__alpha-4;
    stroke: $color__alpha-4;
  }

  &:active,
  &:focus {
    box-shadow: $shadow-size $shadow-color--dark;
  }

  &:hover,
  &:focus {
    background-color: $color__alpha-4;
    border-color: transparent;

    &.noFill {
      color: $color__white;
      fill: $color__white;
      stroke: $color__white;
    }
  }
}

.input-button__mb-input--light {
  background-color: $color__white;
  border-color: transparent;
  color: $color__gray-5;
  fill: $color__gray-5;
  stroke: $color__gray-5;

  &.noFill {
    background-color: transparent;
    border-color: $color__white;
    color: $color__white;
    fill: $color__white;
    stroke: $color__white;
  }

  &:active,
  &:focus {
    box-shadow: $shadow-size $shadow-color--light;
  }

  &:hover,
  &:focus {
    background-color: $color__white;
    border-color: transparent;
    color: $color__gray-4;

    &.noFill {
      color: $color__gray-5;
      fill: $color__gray-5;
      stroke: $color__gray-5;
    }
  }
}

.input-button__mb-input--tertiary {
  background-color: $color__main-tertiary;
  border-color: $color__main-tertiary;
  color: $color__dark-gray-2;
  fill: $color__dark-gray-2;
  stroke: $color__dark-gray-2;

  &.noFill {
    border-color: $color__gray-4;
    background: transparent;
    color: $color__gray-5;
    fill: $color__gray-5;
    stroke: $color__gray-5;
  }

  &:active,
  &:focus {
    box-shadow: $shadow-size $shadow-color--tertiary;
  }

  &:hover,
  &:focus {
    color: $color__dark-gray-2;
    fill: $color__dark-gray-2;
    stroke: $color__dark-gray-2;
    background-color: $color__main-tertiary--active;
    border-color: $color__main-tertiary--active;
  }
}

.input-button__mb-input--disabled {
  background-color: $color__gray-2;
  border-color: $color__gray-2;
  color: $color__dark-gray-1;
  fill: $color__gray-5;
  stroke: $color__gray-5;

  &.noFill {
    border-color: $color__gray-2;
    background: transparent;
    color: $color__gray-5;
    fill: $color__gray-5;
    stroke: $color__gray-5;
  }
}

.input-button__mb-input--primary,
.input-button__mb-input--secondary,
.input-button__mb-input--tertiary,
.input-button__mb-input--danger,
.input-button__mb-input--warning,
.input-button__mb-input--dark,
.input-button__mb-input--light {
  &:hover {
    &.noFill {
      &:active {
        color: $color__alpha-3;
        fill: $color__alpha-3;
        stroke: $color__alpha-3;
      }
    }
  }
  &:active {
    color: $color__alpha-3;
    fill: $color__alpha-3;
    stroke: $color__alpha-3;
  }
}
