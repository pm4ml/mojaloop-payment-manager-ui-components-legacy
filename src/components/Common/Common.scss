@import '../../assets/styles/vars/colors.scss';
@import '../../assets/styles/vars/fonts.scss';

.validation__icon {
  fill: transparent;
}

.validation__icon.validation__icon--invalid {
  height: auto;
  width: auto;
  fill: $color__main-danger;
}

.validation__messages {
  list-style-type: none;
  margin: 0;
  padding: 14px;
  padding-right: 18px;
  background: $color__gray-1;
  box-shadow: 0px 1px 2px $color__alpha-2;
  border-radius: 3px;
  min-width: 60px;
  display: block;
}

.validation__message {
  font-family: $font__primary, $font__primary-fallback;
  letter-spacing: $font__primary-spacing;
  display: flex;
  align-items: center;
  font-size: 12px;
  color: $color__dark-gray-1;
  min-width: 60px;
}

.validation__message--inactive {
  text-decoration: line-through;
  stroke: none;
  color: $color__gray-5;
}

.validation__message__text {
  white-space: nowrap;
}

.validation__message__icon-container {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 16px;
  height: 16px;
  margin-right: 8px;
}

.validation__message__icon--active {
  animation: fadeInValidationMessageIcon 0.2s ease-out;
  fill: $color__main-danger;
  stroke: $color__main-danger;
}

.validation__message__icon--inactive {
  animation: fadeInValidationMessageIcon 0.2s ease-out;
  fill: $color__main-success;
}

.validation__message__icon--unset {
  animation: fadeInValidationMessageIcon 0.2s ease-out;
  height: 6px;
  width: 6px;
  margin: 3px;
  border-radius: 3px;
  display: flex;
  background: #ddd;
}

@keyframes fadeInValidationMessageIcon {
  0% {
    transform: scale(0.5, 0.5);
    opacity: 0.5;
  }

  100% {
    transform: scale(1, 1);
    opacity: 1;
  }
}

.named__icon {
  height: 20px;
  width: 20px;
  border-radius: 20px;
  font-size: 13px;
  display: flex;
  align-items: center;
  justify-content: center;
  background: #022843;
  color: #fff;
  text-transform: uppercase;
  text-decoration: none;

  &:hover {
    text-decoration: none;
  }
}
