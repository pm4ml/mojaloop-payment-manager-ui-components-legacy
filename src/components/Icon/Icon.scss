@import '../../assets/styles/vars/colors.scss';

.el-icon {
  display: flex;
  flex: 0 0 auto;
}

.el-icon--spin {
  animation: spinIcon 1s linear infinite;
}

@keyframes spinIcon {
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
}
