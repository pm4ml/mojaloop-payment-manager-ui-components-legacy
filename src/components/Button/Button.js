import './Button.scss';

import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';

import * as utils from '../../utils/common';
import Icon, { iconSizes } from '../Icon';
import Spinner from '../Spinner';
import Tooltip from '../Tooltip';

class Button extends PureComponent {
  constructor(props) {
    super(props);
    this.onClick = this.onClick.bind(this);
    this.testKey = this.testKey.bind(this);
  }

  onClick(e) {
    e.preventDefault();
    if (this.props.disabled) return;
    if (this.props.onClick) {
      this.props.onClick(e);
    }
  }
  testKey(e) {
    if (e.nativeEvent.keyCode === 9) {
      e.preventDefault();
      utils.focusNextFocusableElement(this.input, !e.nativeEvent.shiftKey);
      return;
    }

    if (e.nativeEvent.keyCode === 13) {
      e.preventDefault();
      this.onClick(e);
    }
  }
  render() {
    const {
      id,
      className,
      style,
      size,
      kind,
      label,
      icon,
      iconPosition,
      noFill,
      disabled,
      pending,
      tooltip,
    } = this.props;

    const isDisabledOrPending = disabled === true || pending === true;
    const iconSize = iconSizes[size];
    const classNames = utils.composeClassNames([
      className,
      'mb-input',
      'input-button__input',
      size === 'xs' && 'input-button__mb-input--extra-small',
      size === 's' && 'input-button__mb-input--small',
      size === 'm' && 'input-button__mb-input--medium',
      size === 'l' && 'input-button__mb-input--large',
      kind === 'primary' && 'input-button__mb-input--primary',
      kind === 'secondary' && 'input-button__mb-input--secondary',
      kind === 'tertiary' && 'input-button__mb-input--tertiary',
      kind === 'success' && 'input-button__mb-input--success',
      kind === 'danger' && 'input-button__mb-input--danger',
      kind === 'warning' && 'input-button__mb-input--warning',
      kind === 'dark' && 'input-button__mb-input--dark',
      kind === 'light' && 'input-button__mb-input--light',
      isDisabledOrPending && 'mb-input--disabled input-button__mb-input--disabled',
      pending && 'mb-input--pending input-button__mb-input--pending',
      noFill && 'noFill',
    ]);

    let iconComponent = null;
    if (pending || icon) {
      iconComponent = (
        <div
          className={`input-button__icon input-button__icon${
            iconPosition === 'left' ? '--left' : '--right'
          }`}
        >
          {pending ? (
            <Spinner color="inherit" size={iconSize} />
          ) : (
            <Icon name={icon} stroke="none" size={iconSize} />
          )}
        </div>
      );
    }

    const leftIcon = iconPosition === 'left' ? iconComponent : null;
    const rightIcon = iconPosition === 'right' ? iconComponent : null;

    const button = (
      <button
        ref={input => {
          this.input = input;
        }}
        id={id}
        style={style}
        className={classNames}
        onKeyDown={this.testKey}
        onClick={this.onClick}
        disabled={isDisabledOrPending}
        label={label}
        kind={kind}
      >
        <div className="input-button__content">
          {leftIcon}
          {label && <span className="input-button__label">{label}</span>}
          {rightIcon}
        </div>
      </button>
    );

    if (tooltip) {
      return (
        <Tooltip label={tooltip} position="top">
          {button}
        </Tooltip>
      );
    }
    return button;
  }
}
Button.propTypes = {
  className: PropTypes.string,
  id: PropTypes.string,
  style: PropTypes.shape(),
  kind: PropTypes.oneOf([
    'primary',
    'secondary',
    'tertiary',
    'success',
    'danger',
    'warning',
    'dark',
    'light',
  ]),
  size: PropTypes.oneOf(['s', 'm', 'l']),
  label: PropTypes.string,
  icon: PropTypes.string,
  iconPosition: PropTypes.oneOf(['left', 'right']),
  noFill: PropTypes.bool,
  disabled: PropTypes.bool,
  pending: PropTypes.bool,
  onClick: PropTypes.func,
  tooltip: PropTypes.string,
};
Button.defaultProps = {
  className: undefined,
  id: undefined,
  style: undefined,
  kind: 'primary',
  size: 'l',
  label: undefined,
  icon: undefined,
  iconPosition: 'left',
  noFill: false,
  disabled: false,
  pending: false,
  onClick: undefined,
  tooltip: undefined,
};

export default Button;
