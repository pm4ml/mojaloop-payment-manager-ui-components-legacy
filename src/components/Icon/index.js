import Icon, { iconSizes } from './Icon';

export default Icon;
export { iconSizes };
