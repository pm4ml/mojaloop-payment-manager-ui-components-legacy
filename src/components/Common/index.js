import './Common.scss';
import '../../icons/mule/check-small.svg';
import '../../icons/mule/close-small.svg';

import PropTypes from 'prop-types';
import React from 'react';

import * as utils from '../../utils/common';
import Button from '../Button';
import Icon, { iconSizes } from '../Icon';
import Row from '../Row';
import Spinner from '../Spinner';
import Tooltip from '../Tooltip';

const Loader = ({ size }) => (
  <div className="mb-input__inner-icon mb-loader">
    <Spinner size={iconSizes[size]} />
  </div>
);

Loader.propTypes = {
  size: PropTypes.oneOf(['s', 'm', 'l']),
};
Loader.defaultProps = {
  size: 'l',
};

const Placeholder = ({ label, size, active }) => {
  // The Placeholder that renders inside an input

  const placeholderClassName = utils.composeClassNames([
    'mb-input__placeholder',
    size === 's' && 'mb-input__placeholder--small',
    size === 'm' && 'mb-input__placeholder--medium',
    size === 'l' && 'mb-input__placeholder--large',
    size === 's' && active && 'mb-input__placeholder--active-small',
    size === 'm' && active && 'mb-input__placeholder--active-medium',
    size === 'l' && active && 'mb-input__placeholder--active-large',
  ]);

  return <label className={placeholderClassName}>{label}</label>;
};

Placeholder.propTypes = {
  label: PropTypes.string,
  active: PropTypes.bool,
  size: PropTypes.oneOf(['s', 'm', 'l']),
};

Placeholder.defaultProps = {
  label: undefined,
  active: false,
  size: 'l',
};

const InnerButton = ({ className, size, kind, onClick, label, disabled, noFill, icon }) => {
  // Internal button used by inputs

  let innerButtonSize;
  if (size === 'l') {
    innerButtonSize = 'm';
  } else if (size === 'm') {
    innerButtonSize = 's';
  } else {
    innerButtonSize = 'xs';
  }

  const innerButtonClassName = utils.composeClassNames([
    className,
    'mb-input__inner-button',
    size === 's' && 'mb-input__inner-button--small',
    size === 'm' && 'mb-input__inner-button--medium',
    size === 'l' && 'mb-input__inner-button--large',
  ]);
  return (
    <Button
      kind={kind}
      size={innerButtonSize}
      className={innerButtonClassName}
      icon={icon}
      noFill={noFill}
      onClick={onClick}
      tabIndex="-1"
      label={label}
      disabled={disabled}
    />
  );
};

InnerButton.propTypes = {
  className: PropTypes.string,
  size: PropTypes.oneOf(['s', 'm', 'l']),
  kind: PropTypes.string,
  icon: PropTypes.string,
  onClick: PropTypes.func,
  label: PropTypes.string,
  disabled: PropTypes.bool,
};

InnerButton.defaultProps = {
  className: undefined,
  size: 'l',
  kind: 'primary',
  icon: undefined,
  onClick: undefined,
  label: undefined,
  disabled: false,
};

const ActiveValidationIcon = () => (
  <Icon className="validation__message__icon--active" name="close-small" size={12} />
);
const InactiveValidationIcon = () => (
  <Icon className="validation__message__icon--inactive" name="check-small" size={14} />
);
const UnsetValidationIcon = () => <div className="validation__message__icon--unset" />;

const ValidationIcon = ({ active }) => {
  if (active === true) {
    return <ActiveValidationIcon />;
  } else if (active === false) {
    return <InactiveValidationIcon />;
  }
  return <UnsetValidationIcon />;
};

const ValidationMessage = ({ message, active }) => (
  <Row>
    <li
      className={`validation__message ${active === false ? 'validation__message--inactive' : ''}`}
    >
      <div className="validation__message__icon-container">
        <ValidationIcon active={active} />
      </div>
      <span className="validation__message__text">{message}</span>
    </li>
  </Row>
);

ValidationMessage.propTypes = {
  message: PropTypes.string,
  active: PropTypes.bool,
};
ValidationMessage.defaultProps = {
  message: undefined,
  active: undefined,
};

const ValidationMessages = ({ messages }) => {
  let validationMessageList = null;
  if (messages.length) {
    validationMessageList = messages.map(({ message, active }, i) => (
      <ValidationMessage key={i.toString()} message={message} active={active} />
    ));
  }
  return <ul className="validation__messages">{validationMessageList}</ul>;
};

ValidationMessages.propTypes = {
  messages: PropTypes.arrayOf(PropTypes.shape(ValidationMessage.propTypes)),
};

ValidationMessages.defaultProps = {
  messages: [],
};

const InvalidIcon = ({ size }) => (
  <Icon
    size={iconSizes[size]}
    name="warning-sign"
    className="mb-input__inner-icon validation__icon validation__icon--invalid "
    tooltip="Invalid"
    tooltipDelay={500}
    tooltipKind="error"
  />
);

const ValidationWrapper = ({ messages, active, children }) => (
  <Tooltip
    position="right"
    kind="neutral"
    custom
    delay={500}
    content={<ValidationMessages messages={messages} />}
    forceVisibility={active === true && messages && messages.length > 0}
    showOnHover={false}
  >
    {children}
  </Tooltip>
);

const NamedIcon = ({ namedIconTitle }) => <div className="named__icon">{namedIconTitle[0]}</div>;

NamedIcon.propTypes = {
  namedIconTitle: PropTypes.string,
};
NamedIcon.defaultProps = {
  namedIconTitle: undefined,
};

export { Loader, Placeholder, InnerButton, ValidationWrapper, InvalidIcon, NamedIcon };
