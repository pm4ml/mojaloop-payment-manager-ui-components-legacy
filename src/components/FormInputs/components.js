import './FormInput.scss';
import '../../icons/mule/forum-small.svg';

import React, { PureComponent } from 'react';

import * as utils from '../../utils/common';
import Button from '../Button';
import Icon from '../Icon';
import Tooltip from '../Tooltip';

export const FieldInfoOverlay = ({ assignRef, title, description, url, required }) => (
  <div className="forminput__field-info" ref={assignRef}>
    <div className="forminput__field-info__title-container">
      <div className="forminput__field-info__title">{title}</div>
      {required && (
        <React.Fragment>
          <div className="forminput__field-info__required-icon">
            <Icon size={20} name="info-small" className="forminput__label-icon" fill="#333" />
          </div>
          <div className="forminput__field-info__required-label"> required </div>
        </React.Fragment>
      )}
    </div>
    <div className="forminput__field-info__content">
      <p>{description}</p>
    </div>
    {url && (
      <div className="forminput__field-info__url">
        <a href={url}>{url}</a>
      </div>
    )}
  </div>
);

export class FieldInfo extends PureComponent {
  constructor() {
    super();
    this.onClick = this.onClick.bind(this);
    this.open = this.open.bind(this);
    this.close = this.close.bind(this);
    this.closeIfClickingOutside = this.closeIfClickingOutside.bind(this);
    this.state = {
      open: false,
    };
  }
  componentWillUnmount() {
    this.close();
  }
  onClick() {
    if (this.state.open) {
      this.close();
    } else {
      this.open();
    }
  }
  open() {
    this.area = React.createRef();
    this.setState(
      {
        open: true,
      },
      () => {
        document.addEventListener('click', this.closeIfClickingOutside);
      },
    );
  }
  close() {
    this.setState(
      {
        open: false,
      },
      () => {
        this.area = null;
        document.removeEventListener('click', this.closeIfClickingOutside);
      },
    );
  }
  closeIfClickingOutside(evt) {
    if (!this.area.current) {
      return;
    }
    if (!this.area.current.contains(evt.target)) {
      this.close();
    }
  }
  render() {
    const { iconSize, title, required, description, url } = this.props;
    let content = null;
    if (this.state.open) {
      content = (
        <FieldInfoOverlay
          assignRef={this.area}
          title={title}
          description={description}
          url={url}
          required={required}
        />
      );
    }
    const fieldInfoClassName = utils.composeClassNames([
      'forminput__field-info__icon',
      this.state.open && 'forminput__field-info__icon--active',
    ]);

    return (
      <Tooltip
        forceVisibility={this.state.open}
        showOnHover={false}
        content={content}
        custom
        position="right"
        align="start"
      >
        <Icon
          name="forum-small"
          onClick={this.onClick}
          size={iconSize}
          className={fieldInfoClassName}
        />
      </Tooltip>
    );
  }
}

export const Label = ({ size, label, required, complete, description, url }) => {
  if (!label) {
    return null;
  }

  const className = utils.composeClassNames([
    'forminput__label-box',
    size === 's' && 'forminput__label-box--small',
    size === 'm' && 'forminput__label-box--medium',
    size === 'l' && 'forminput__label-box--large',
  ]);
  const iconSizes = {
    s: 11,
    m: 12,
    l: 14,
  };
  const iconSize = iconSizes[size];
  return (
    <div className={className}>
      {required && (
        <Icon
          size={iconSize}
          name="info-small"
          className="forminput__label-icon"
          fill={complete ? '#39f' : '#f93'}
          tooltip={complete ? '' : 'This is a required field'}
        />
      )}
      <label>{label}</label>
      {description && (
        <FieldInfo
          iconSize={iconSize}
          title={label}
          description={description}
          url={url}
          required={required}
        />
      )}
    </div>
  );
};

export const InlineButton = ({ visible, isLocked, isDisabled, onClick, label }) => {
  if (!visible || isLocked || isDisabled) {
    return null;
  }

  return (
    <Button
      size="m"
      noFill
      kind="secondary"
      className="forminput__inline-button"
      onClick={onClick}
      disabled={isDisabled}
      label={label}
    />
  );
};

export const LockedIcon = ({ locked }) => {
  if (!locked) {
    return null;
  }
  return (
    <div className="forminput__inline-icon">
      <Icon
        name="lock-small"
        size={20}
        fill="#999"
        style={{ marginLeft: '10px' }}
        tooltip="this field is locked"
      />
    </div>
  );
};

export const InfoMessage = ({ message }) => (
  <div className="forminput__input-message">{message && <Tooltip>{message}</Tooltip>}</div>
);
