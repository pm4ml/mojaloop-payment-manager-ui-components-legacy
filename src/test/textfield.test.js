import { mount, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

import Button from '../components/Button';
import { InvalidIcon, Loader, Placeholder, ValidationWrapper } from '../components/Common';
import Icon from '../components/Icon';
import TextField from '../components/TextField';

it('renders the textfield', () => {
  const wrapper = shallow(<TextField />);
  expect(wrapper.find('div.input-textfield__component')).toHaveLength(1);
});

it('renders the textfield value', () => {
  const wrapper = shallow(<TextField value="test-value" />);
  expect(wrapper.find('input').prop('value')).toBe('test-value');
});

it('renders the placeholder', () => {
  const wrapper = shallow(<TextField placeholder="test-textfield" />);
  expect(wrapper.find(Placeholder)).toHaveLength(1);
  expect(wrapper.find(Placeholder).prop('label')).toBe('test-textfield');
});

it('renders the validation wrapper', () => {
  const wrapper = shallow(<TextField placeholder="test-Select" />);
  expect(wrapper.find(ValidationWrapper)).toHaveLength(1);
});

it('applies the prop className', () => {
  const wrapper = shallow(<TextField className="test" />);
  expect(wrapper.find('.test')).toHaveLength(1);
});

it('renders the prop id', () => {
  const wrapper = shallow(<TextField id="testTextFieldId" />);
  expect(wrapper.find('#testTextFieldId')).toHaveLength(1);
});

it('renders the disabled state', () => {
  const wrapper = shallow(<TextField disabled />);
  expect(wrapper.find('input')).toHaveLength(1);
  expect(wrapper.find('input').prop('disabled')).toBe(true);
});

it('renders the icon', () => {
  const wrapper = mount(<TextField icon="deploy" />);
  expect(wrapper.find(Icon).prop('name')).toBe('deploy');
});

it('renders the pending state', () => {
  const wrapper = shallow(<TextField pending />);
  expect(wrapper.find(Loader)).toHaveLength(1);
});

it('renders the invalid state', () => {
  const wrapper = shallow(<TextField invalid />);
  expect(wrapper.find(InvalidIcon)).toHaveLength(1);
});

it('renders the required state', () => {
  const wrapper = shallow(<TextField required />);
  expect(wrapper.find('.mb-input--required')).toHaveLength(1);
});

it('renders the large, medium, small sizes', () => {
  const sizes = {
    large: 'l',
    medium: 'm',
    small: 's',
  };
  Object.entries(sizes).forEach(([name, size]) => {
    const wrapper = shallow(<TextField size={size} />);
    const className = `mb-input--${name}`;
    expect(wrapper.find('.mb-input').hasClass(className)).toBeTruthy();
  });
});

it('renders the password kind and clicks the toggle icon', () => {
  const wrapper = mount(<TextField type="password" />);
  expect(wrapper.find(Icon)).toHaveLength(1);
  expect(wrapper.find(Icon).prop('name')).toBe('toggle-visible');
  expect(wrapper.find('input').prop('type')).toBe('password');
  wrapper.find(Icon).simulate('click');
  expect(wrapper.find(Icon).prop('name')).toBe('toggle-invisible');
  expect(wrapper.find('input').prop('type')).toBe('text');
});

it('renders the number kind', () => {
  const wrapper = mount(<TextField type="number" />);
  expect(wrapper.find('input').prop('type')).toBe('number');
});

it('renders the inner button', () => {
  const mockEvent = jest.fn();
  const wrapper = mount(<TextField onButtonClick={mockEvent} buttonText="inner-button-text" />);
  expect(wrapper.find(Button).text()).toBe('inner-button-text');
});

it('renders the inner button disabled', () => {
  const mockEvent = jest.fn();
  const wrapper = mount(
    <TextField onButtonClick={mockEvent} buttonText="inner-button-text" buttonDisabled />,
  );
  expect(wrapper.find(Button).prop('disabled')).toBe(true);
});

it('renders the inner button with kind', () => {
  const mockEvent = jest.fn();
  const wrapper = mount(
    <TextField onButtonClick={mockEvent} buttonText="inner-button-text" buttonKind="secondary" />,
  );
  expect(wrapper.find(Button).prop('kind')).toBe('secondary');
});

it('renders and clicks the inner button', () => {
  const mockEvent = jest.fn();
  const wrapper = mount(<TextField onButtonClick={mockEvent} buttonText="inner-button-text" />);
  expect(mockEvent).not.toHaveBeenCalled();
  wrapper.find(Button).simulate('click');
  expect(mockEvent).toHaveBeenCalled();
});

it('triggers onfocus when focused', () => {
  const mockEvent = jest.fn();
  const wrapper = mount(<TextField onFocus={mockEvent} />);
  expect(mockEvent).not.toHaveBeenCalled();
  wrapper.find('input').simulate('focus');
  expect(mockEvent).toHaveBeenCalled();
});

it('triggers onBlur when blurred', () => {
  const mockEvent = jest.fn();
  const wrapper = mount(<TextField onBlur={mockEvent} />);
  expect(mockEvent).not.toHaveBeenCalled();
  wrapper.find('input').simulate('blur');
  expect(mockEvent).toHaveBeenCalled();
});

it('triggers onChange when changed', () => {
  const mockEvent = jest.fn();
  const wrapper = mount(<TextField onChange={mockEvent} />);
  expect(mockEvent).not.toHaveBeenCalled();
  wrapper.find('input').simulate('change', { target: { value: 'test' } });
  expect(mockEvent).toHaveBeenCalledWith('test');
});

it('triggers onChange with a number value when prop type is number', () => {
  const mockEvent = jest.fn();
  const wrapper = mount(<TextField onChange={mockEvent} type="number" />);
  expect(mockEvent).not.toHaveBeenCalled();
  wrapper.find('input').simulate('change', { target: { value: '12' } });
  expect(mockEvent).toHaveBeenCalledWith(12);
});

describe('tests the textfield cards', () => {
  it('identifies the cardable cards of a given value automatically', () => {
    const wrapper = mount(<TextField value="[test]value" tokenDelimiters="[]" />);
    const tokens = wrapper.state('tokens');
    const [token] = tokens;
    expect(tokens).toHaveLength(1);
    expect(token.isCardable).toBe(true);
    expect(token.isPartiallyCardable).toBe(false);
    expect(token.word).toBe('[test]');
  });

  it('identifies the partially cards of a given value automatically', () => {
    const wrapper = mount(<TextField value="[partial[test]value" tokenDelimiters="[]" />);
    const tokens = wrapper.state('tokens');
    const [partial, token] = tokens;
    expect(tokens).toHaveLength(2);
    expect(partial.isCardable).toBe(false);
    expect(partial.isPartiallyCardable).toBe(true);
    expect(partial.word).toBe('[partial');
    expect(token.isCardable).toBe(true);
    expect(token.isPartiallyCardable).toBe(false);
    expect(token.word).toBe('[test]');
  });

  it('identifies the escaped cards of a given value automatically', () => {
    const wrapper = mount(<TextField value="\[test]value" tokenDelimiters="[]" />);
    const tokens = wrapper.state('tokens');
    expect(tokens).toHaveLength(0);
  });

  it('renders the ValueTokens components', () => {
    const tokensResult = [
      {
        value: 'test',
        available: true,
        replaced: 'I was a value',
      },
      {
        value: 'missing',
        available: false,
        replaced: '',
      },
    ];
    const wrapper = mount(
      <TextField value="[test][missing]foobar" tokenDelimiters="[]" tokens={tokensResult} />,
    );
    const ValueTokens = wrapper.find('ValueToken');
    const Test = ValueTokens.at(0);
    const Missing = ValueTokens.at(1);

    expect(ValueTokens).toHaveLength(2);
    expect(Test.prop('isCardable')).toBe(true);
    expect(Test.prop('isInvalid')).toBe(false);
    expect(Test.prop('isSelected')).toBe(false);
    expect(Test.prop('word')).toBe('[test]');
    expect(Missing.prop('isCardable')).toBe(true);
    expect(Missing.prop('isInvalid')).toBe(true);
    expect(Missing.prop('isSelected')).toBe(false);
    expect(Missing.prop('word')).toBe('[missing]');
  });
});

// Snapshot
it('renders the textfield correctly when multiple props are set', () => {
  const wrapper = shallow(<TextField value="test-value" id="test-id" />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
