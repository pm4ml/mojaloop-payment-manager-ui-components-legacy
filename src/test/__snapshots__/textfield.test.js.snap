// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the textfield correctly when multiple props are set 1`] = `
<ValidationWrapper
  active={false}
  messages={Array []}
>
  <div
    className="input-textfield__component mb-input mb-input__borders mb-input__background mb-input__shadow mb-input--large"
    onClick={[Function]}
    role="presentation"
    style={Object {}}
  >
    <div
      className="mb-input__content input-textfield__content"
    >
      <div
        className="input-textfield__value__tokens"
      >
        <input
          autoComplete="off"
          autoFocus={false}
          className="mb-input__input input-textfield__value"
          disabled={false}
          id="test-id"
          key="input"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          type="text"
          value="test-value"
        />
      </div>
    </div>
  </div>
</ValidationWrapper>
`;
